test/token/FTCVesting.spec.ts

```diff
@@ -0,0 +1,293 @@
+        subjectVestingStart,
+        subjectVestingCliff,
+        subjectVestingEnd,
+      );
+    });
+
+    async function subject(): Promise<ContractTransaction> {
+      return subjectFtcVesting.connect(recipient.wallet).setRecipient(newRecipient.address);
+    }
+
+    it("should set new recipient address", async () => {
+      await subject();
+
+      const newRecipientAddress = await subjectFtcVesting.recipient();
+
+      expect(newRecipientAddress).to.eq(newRecipient.address);
+    });
+  });
+
+  describe("#setTreasury", async () => {
+    let subjectVestingAmount: BigNumber;
+    let subjectVestingStart: BigNumber;
+    let subjectVestingCliff: BigNumber;
+    let subjectVestingEnd: BigNumber;
+    let subjectFtcVesting: FTCVesting;
+    let newTreasury: Account;
+
+    beforeEach(async () => {
+      const now = await getLastBlockTimestamp();
+      newTreasury = await getRandomAccount();
+      subjectVestingStart = now;
+      subjectVestingCliff = now.add(60 * 60 * 24 * 183);
+      subjectVestingEnd = now.add(60 * 60 * 24 * 547);
+      subjectVestingAmount = ether(100);
+
+      subjectFtcVesting = await deployer.token.deployFtcVesting(
+        index.address,
+        recipient.address,
+        treasury.address,
+        subjectVestingAmount,
+        subjectVestingStart,
+        subjectVestingCliff,
+        subjectVestingEnd,
+      );
+    });
+
+    async function subject(): Promise<ContractTransaction> {
+      return subjectFtcVesting.connect(treasury.wallet).setTreasury(newTreasury.address);
+    }
+
+    it("should set new treasury address", async () => {
+      const currentTreasuryAddress = await subjectFtcVesting.treasury();
+
+      expect(currentTreasuryAddress).to.eq(treasury.address);
+
+      await subject();
+
+      const newTreasuryAddress = await subjectFtcVesting.treasury();
+
+      expect(newTreasuryAddress).to.eq(newTreasury.address);
+    });
+  });
+
+  describe("#claim", async () => {
+    let subjectVestingAmount: BigNumber;
+    let subjectVestingStart: BigNumber;
+    let subjectVestingCliff: BigNumber;
+    let subjectVestingEnd: BigNumber;
+    let subjectFtcVesting: FTCVesting;
+
+    beforeEach(async () => {
+      const now = await getLastBlockTimestamp();
+      subjectVestingStart = now.sub(60 * 60 * 24 * 185);
+      subjectVestingCliff = now;
+      subjectVestingEnd = now.add(60 * 60 * 24 * 546);
+      subjectVestingAmount = ether(1);
+
+      subjectFtcVesting = await deployer.token.deployFtcVesting(
+        index.address,
+        recipient.address,
+        treasury.address,
+        subjectVestingAmount,
+        subjectVestingStart,
+        subjectVestingCliff,
+        subjectVestingEnd,
+      );
+
+      await index.transfer(subjectFtcVesting.address, ether(1000));
+    });
+
+    async function subject(): Promise<ContractTransaction> {
+      return await subjectFtcVesting.connect(recipient.wallet).claim();
+    }
+
+    it("should make a claim", async () => {
```

Comment:
Can we split this into two tests? One for a partial claim and one for a full claim. Typically you only want to call subject() once per test.